### ISSUE TITLE
WRN_UnreferencedVarAssg "The variable '...' is assigned but its value is never used" is not reported for out variables

### ISSUE THREAD

**Issue by gafter** [dotnet/roslyn#12225]
```
    void Test3()
    {
        M(out var x3);
        var x4 = 12;
    }
```

Observed:

```
    // (11,13): warning CS0219: The variable 'x4' is assigned but its value is never used
    //         var x4 = 12;
    Diagnostic(ErrorCode.WRN_UnreferencedVarAssg, "x4").WithArguments("x4").WithLocation(11, 13)
```

Expected:
WRN_UnreferencedVarAssg should also be reported for x3.

@AlekseyTs You can resolve this when you resolve #9021, and I expect with the same outcome.

**Comment by AlekseyTs** [comment]
This behavior is consistent with behavior for all locals used as ```out``` arguments.
```
    void Test3()
    {
        int x3;
        M(out x3);
    }

    void M(out int x)
    {
        x = 0;
    }
```

Observed: No warnings or errors.

**Comment by gafter** [comment]
I understand that you want an explicit LDM decision for these two issues (even though these are not language issues). Personally I agree that the feature designer did the right thing on them.

**Comment by gafter** [comment]
Due to [changes recently proposed by the LDM](https://github.com/dotnet/roslyn/issues/14794), if we implement wildcards in C#7 we will likely want this warning.

**Comment by gafter** [comment]
The LDM met yesterday and confirmed that we do not intend there to be new warnings for these scenarios, though IDEs and other tools may want to produce helpful style guidance.
